### PR TITLE
Connectors Permissions UI

### PR DESCRIPTION
**Pull request by spolu**
This PR implements a basic UI to list the top-level resources a managed data source has access to for each platform. It's quite limited (does not allow traversing the resources despite it being possible for Notion)

I moved the Edit permission button inside the modal but this one is still walled by the query parameter `?updatePermissions=enabled`. I tried to converged towards visuals in the Product XP Figma, but run into issues using sparkle icons from `front` :/

Will try to follow-up on that in another PR.

This should be functional and a step in the right direction but will need a more thorough design pass.

Screenshot from 2023-07-24 16-09-15

### CODE REVIEW

**Comment by spolu** [comment]
Will try to move the modal back up a notch

**Comment by spolu** [comment]
cc @gabhubert for wording if you have anything better in mind?

**Comment by spolu** [comment]
cc @Duncid not awesome, but not horrible either. I went with a Close button out of consistency with the rest of the app and we can move that to being a slide panel coming from the right soon

File: front/pages/w/[wId]/ds/[name]/settings.tsx [hunk 1]
```diff
@@ -585,6 +596,19 @@ function ManagedDataSourceSettings({
...
       )}
+      {connector && (
+        <ConnectorPermissionsModal
+          owner={owner}
+          connector={connector}
+          dataSource={dataSource}
+          isOpen={showPermissionModal}
+          setOpen={setShowPermissionModal}
+          onEditPermission={() => {
+            console.log("Update Permnission");
```
Comment (philipperolet) [inline front/pages/w/[wId]/ds/[name]/settings.tsx:607]: this was here only for dev no? can be removed? (if not, typo :) )

File: front/pages/w/[wId]/ds/[name]/settings.tsx [hunk 2]
```diff
@@ -473,6 +474,15 @@ function ManagedDataSourceStatus({
   fetchConnectorError: boolean;
   connector: ConnectorType | null;
 }) {
+  const [syncrhonizedTimeAgo, setSyncrhonizedTimeAgo] = useState<string | null>(
```
Comment (philipperolet) [inline front/pages/w/[wId]/ds/[name]/settings.tsx:477]: typo sync**hro** not **rho**

File: front/pages/api/w/[wId]/data_sources/[name]/managed/permissions.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    session,
+    req.query.wId as string
+  );
+
+  const owner = auth.workspace();
+  if (!owner) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "data_source_not_found",
```
Comment (philipperolet) [inline front/pages/api/w/[wId]/data_sources/[name]/managed/permissions.ts:30]: copy-paste mistake or intentional? I saw "workspace not found in some other api calls

**Review by philipperolet — DISMISSED** [review]
LGTM :+1:  3 nits / typos

File: front/pages/api/w/[wId]/data_sources/[name]/managed/permissions.ts (hunk 3 above)
Comment (spolu) [reply front/pages/api/w/[wId]/data_sources/[name]/managed/permissions.ts:30]: this is just a reorder we do use `data_source_not_found` :+1:

File: front/pages/w/[wId]/ds/[name]/settings.tsx (hunk 2 above)
Comment (spolu) [reply front/pages/w/[wId]/ds/[name]/settings.tsx:477]: :+1:

**Comment by spolu** [comment]
Thank you! Addressed your comments. Also removed sparkle for now since I didn't get to make it work yet :/

File: front/components/ConnectorPermissionsModal.tsx [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+                    >
+                      <Cog6ToothIcon className="mr-2 h-5 w-5" />
+                      Edit permissions
+                    </Button>
+                  </div>
+                </div>
+                <div>
+                  <div className="mt-16">
+                    <div className="px-2 text-sm text-gray-500">
+                      Top-level resources accessible by this managed data
```
Comment (gabhubert) [inline front/components/ConnectorPermissionsModal.tsx:189]: "Dust has access to the following top-level $ConnectResourceType{s}:"

**Review by gabhubert — COMMENTED** [review]
2 small things I think

**Comment by spolu** [comment]
@gabhubert ack will address in a follow-up PR

**Pull request merged**